### PR TITLE
Green Uop unary check

### PR DESCRIPTION
**Pull request by Qazalin**
#2787
- linearizer should fallback to float32 for imagef https://github.com/tinygrad/tinygrad/actions/runs/7229870390/job/19701282549?pr=2787#step:14:302
- skipping non-unary ops for now (still handling the dtypes in the uop function). To unblock binary ops, we need two things:
1. `add`ing two bools (eg. CMPLT results) should autocast to int, then get rid of logic and in llvm: https://github.com/tinygrad/tinygrad/blob/master/tinygrad/renderer/llvmir.py#L17
https://github.com/tinygrad/tinygrad/actions/runs/7229870390/job/19701282273?pr=2787#step:14:2417
2. remove CMPLT f32() hack in webgpu
https://github.com/tinygrad/tinygrad/blob/master/tinygrad/renderer/cstyle.py#L340

### CODE REVIEW

**Comment by chenyuxyz** [comment]
cool! if you want you can do the unary and binary check. I am working on the where one https://github.com/chenyuxyz/tinygrad/pull/54

**Comment by Qazalin** [comment]
sg - do you think we can merge this one for unaries and I'll work on binary next? @chenyuxyz

**Comment by Qazalin** [comment]
gotta revert again - i'll figure out why

**Comment by chenyuxyz** [comment]
we should also add more fuzz / kernel search test to CI to increase kernel variety

File: tinygrad/codegen/linearizer.py [hunk 1]
```diff
@@ -64,6 +64,10 @@ def get_reduce_acc(self, op, dtype:DType):
     AndNode: lambda self,ops,ctx:
       functools.reduce(lambda a,b: ctx.uop_alu_idx(a, b, ops, ctx, BinaryOps.MUL, dtype=dtypes.bool), self.nodes[1:], self.nodes[0].render(ops,ctx)) }
 
+  def get_uop_dtype(self, dtype: DType, amt=1):
+    if isinstance(dtype, ImageDType): dtype = dtypes.float
```
Comment (chenyuxyz) [inline tinygrad/codegen/linearizer.py:68]: also should this be `dtype.base` and not hard coded to float here?

Comment (Qazalin) [reply tinygrad/codegen/linearizer.py:68]: oh cool we finally have that

**Pull request merged**